### PR TITLE
Make kapt annotation processor almost incremental

### PR DESCRIPTION
**Pull request by emartynov**
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring

## :scroll: Description
Changes:
* Update gradle to latest in hope for the better changes.
* Use ~~AGP 3.5 beta,~~ latest Dagger and latest Glide deps that support incremental annotation processing.
* Configure Dagger to be incremental
* Configure Kapt to use cache

## :bulb: Motivation and Context
It was part of the lighting talk where I showed how to improve build incrementality. 

## :green_heart: How did you test it?
Run assembleDebug and check it still runs on a device.

## :pencil: Checklist

- [x] I ran `./gradlew spotlessApply` before submitting the PR
- [x] I reviewed submitted code

## :crystal_ball: Next steps
Review and merge if agreed

## :camera_flash: Screenshots / GIFs

There is no UI changes

### CODE REVIEW

**Comment by emartynov** [comment]
That is the super unfortunate thing. I have tests passing locally. How do you get a test run XML to get more details about failure?

File: core/src/main/java/io/plaidapp/core/util/glide/ImageSpanTarget.kt [hunk 1]
```diff
@@ -36,8 +37,7 @@ import java.lang.ref.WeakReference
 class ImageSpanTarget(
     textView: TextView,
     private val loadingSpan: ImageLoadingSpan
-) : SimpleTarget<Bitmap>() {
-
+) : CustomViewTarget<TextView, Bitmap>(textView) {
```
Comment (nickbutcher) [inline core/src/main/java/io/plaidapp/core/util/glide/ImageSpanTarget.kt]: This change looks unrelated to the rest of this PR, please remove.

File: gradlew [hunk 2]
```diff
@@ -1,5 +1,21 @@
 #!/usr/bin/env sh
 
+#
```
Comment (nickbutcher) [inline gradlew]: Was this intentional? We don't put headers on gradle's generated files. Please remove.

File: gradlew.bat [hunk 3]
```diff
@@ -1,3 +1,19 @@
+@rem
```
Comment (nickbutcher) [inline gradlew.bat]: Same, please remove.

File: repositories.gradle [hunk 4]
```diff
@@ -50,6 +50,7 @@ repositories {
             includeGroup "org.ow2.asm"
             includeGroup "org.sonatype.oss"
             includeGroup "org.xerial"
+            includeGroup "net.ltgt.gradle.incap"
```
Comment (nickbutcher) [inline repositories.gradle]: What is this??

File: core/src/main/java/io/plaidapp/core/util/glide/ImageSpanTarget.kt (hunk 1 above)
Comment (emartynov) [reply core/src/main/java/io/plaidapp/core/util/glide/ImageSpanTarget.kt]: Nick, it is a new lint warning about depreciation that is coming from Glide update

File: gradlew (hunk 2 above)
Comment (emartynov) [reply gradlew]: I usually update gradle with the command `./gradlew wrapper --gradle-version=x`. This will modify files' content in the .gradle folder and I trust what gradle team thinks is important in this folder.

File: repositories.gradle (hunk 4 above)
Comment (emartynov) [reply repositories.gradle]: Ah, I need to verify. This dependency was needed for the AGP update. Since I failed to update AGP and reverted that change I should also revert this one.

File: core/src/main/java/io/plaidapp/core/util/glide/ImageSpanTarget.kt (hunk 1 above)
Comment (emartynov) [reply core/src/main/java/io/plaidapp/core/util/glide/ImageSpanTarget.kt]: My default DOD is that my changes don't introduce any new compile, lint or other checkers warnings.

File: gradlew (hunk 2 above)
Comment (emartynov) [reply gradlew]: Do you still want me to remove it?

Comment (nickbutcher) [reply gradlew]: Just verified that this is coming from gradle, I was concerned that your IDE was auto-adding a copyright header as the file had been changed.

File: repositories.gradle (hunk 4 above)
Comment (emartynov) [reply repositories.gradle]: Just checked and remember it now. This dependency is needed for a new gradle compiler. If I remove that line I get the error:

```
Could not determine the dependencies of task ':core:kaptDebugKotlin'.
> Could not resolve all task dependencies for configuration ':core:kapt'.
   > Could not find net.ltgt.gradle.incap:incap:0.2.
     Required by:
         project :core > com.google.dagger:dagger-compiler:2.23.2
```

Comment (nickbutcher) [reply repositories.gradle]: 👍

**Comment by nickbutcher** [comment]
Thanks for the contribution!

**Comment by emartynov** [comment]
It could be part of #718 since improves incremental build times.
